urbanopt/geojson-modelica-translator
# Adds method to process modelica results

Objective : 

- add uo des cli call and methods to query results from modelica models 
- these results are required for the REOPT GHP LCCA analysis that will be run using the urbanopt SDK 
- this cli call will be run in the following sequence : 
1) create modelica model with GHP
2) run thermal network to size GHP
3) run modelica model 
4) query results using this cli call 
5) run reopt GHP LCCA analysis using urbanopt cli 

This cli call takes the modelica folder created and run previously as an argument. It then reads the .mat file and queries timeseries results for heating, cooling electricity consumption of heat pumps, ets pump electricity consumption, pump associated with building electricity consumption loop pump electricity consumption and heat pump sizing for the GHP network. 

This uses methods from the modelica buildingspy library to query results. https://simulationresearch.lbl.gov/modelica/buildingspy/

To test : 


Use `uo_des des-process `<modelica_model_name>`

Run `poetry run pytest tests poetry run pytest .\tests\geojson_modelica_translator\test_results_ghp.py`

Check the "geojson-modelica-translator\tests\geojson_modelica_translator\data\modelica_5\modelica_5.Districts.DistrictEnergySystem_results\modelica_5.Districts.DistrictEnergySystem_result.csv" file created with results queried for REopt GHP at 15 minute intervals.
